source/adios2/toolkit/format/bp/bp3/BP3Deserializer.tcc

```diff
@@ -581,11 +581,11 @@ void BP3Deserializer::PostDataRead(
             intersectStart[d] += blockInfo.MemoryStart[d];
             blockStart[d] += blockInfo.MemoryStart[d];
         }
-        helper::NdCopy<T>(
-            m_ThreadBuffers[threadID][0].data(), intersectStart, intersectCount,
-            true, true, reinterpret_cast<char *>(blockInfo.Data),
-            intersectStart, intersectCount, true, true, intersectStart,
-            blockCount, memoryStart, blockInfo.MemoryCount, false);
+        helper::NdCopy(m_ThreadBuffers[threadID][0].data(), intersectStart,
+                       intersectCount, true, true,
+                       reinterpret_cast<char *>(blockInfo.Data), intersectStart,
+                       intersectCount, true, true, sizeof(T), intersectStart,
+                       blockCount, memoryStart, blockInfo.MemoryCount, false);
     }
```

Comment:
Oh I didn't know that BP3 has been using this function too. Was it recently modified to use it?

Reply:
https://github.com/ornladios/ADIOS2/commit/b6758759fca886bd2dcf28ac374bbead89209fc6

Reply:
Then it's easy. There must be many tests using it already besides DataMan. In this case as long as all tests pass it should be safe to merge it.